paasta_tools/subscribe_mesos_master.py

```diff
@@ -0,0 +1,247 @@
+                while True:
+                    _size = await resp.content.readline()
+                    if not _size:
+                        break
+                    size = int(_size)
+                    record = await resp.content.readexactly(size)
+                    yield json.loads(record)
+
+    def determine_service_instance(self, framework_name, status):
+        executor_id = status.get("executor_id", {}).get("value")
+        task_id = status.get("task_id", {}).get("value")
+
+        if framework_name.startswith("marathon"):
+            return marathon_tools.parse_service_instance_from_executor_id(
+                executor_id or task_id
+            )
+        elif framework_name.startswith("tron"):
+            return tron_tools.decompose_executor_id(executor_id)[:2]
+        elif framework_name.startswith("paasta-remote "):
+            # sorta gross, but it's the same format.
+            return marathon_tools.parse_service_instance_from_executor_id(executor_id)
+        else:
+            raise UnknownFrameworkError(
+                f"don't know how to parse task IDs for framework {framework_name}"
+            )
+
+    def skip_updates_from_framework(self, framework_name) -> bool:
+        if framework_name.startswith("jupyterhub_"):
+            return True
+        if framework_name.startswith("jenkins"):
+            return True
+        if framework_name.startswith("paasta_spark_run"):
+            return True
+        return False
+
+    def handler_task_updated(self, event):
+        task_updated = event["task_updated"]
+        try:
+            del task_updated["status"]["data"]
+        except KeyError:
+            pass
+
+        framework_name = self.framework_id_to_name[
+            task_updated["framework_id"]["value"]
+        ]
+        if self.skip_updates_from_framework(framework_name):
+            return
+
+        service, instance = self.determine_service_instance(
+            framework_name, task_updated["status"]
+        )
+        git_sha = self.framework_id_task_id_to_git_sha.get(
+            (
+                task_updated["framework_id"]["value"],
+                task_updated["status"]["task_id"]["value"],
+            )
+        )
+
+        self.log_task_updated(
+            service=service,
+            instance=instance,
+            git_sha=git_sha,
+            task_updated=task_updated,
+        )
+
+    def log_task_updated(self, service, instance, git_sha, task_updated):
+        message = {
+            "type": "mesos_task_updated",
+            "is_terminal": task_updated["state"] in mesos_tools.TERMINAL_STATES,
+            "interpretation": interpret_task_updated(task_updated),
+            "git_sha": git_sha,
+            "task_updated": task_updated,
+        }
+        utils._log(
+            service=service,
+            instance=instance,
+            component=LOG_COMPONENT,
+            cluster=self.cluster,
+            line=json.dumps(message),
```

Comment:
We don't normally put json in here, right? `_log` *already* puts json onto scribe, so this is json in json?

I guess you want that so that m-f-d or whatever can read this programmatically. Hmm...

Reply:
Yeah this is json-in-json. _log takes a `str` for line. In practice I could _probably_ just pass a dict for message, but that might break something downstream that assumes it's a string.